src/test/java/com/artipie/asto/SubStorageTest.java

```diff
@@ -106,4 +106,73 @@ void checksExistence(final String pref) {
             new SubStorage(Key.ROOT, this.asto).exists(new Key.From(prefix, "any.txt")).join()
         );
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"my-project", "com/example"})
+    void savesValue(final String pref) {
+        final Key prefix = new Key.From(pref);
+        final byte[] data = "some data".getBytes(StandardCharsets.UTF_8);
+        final SubStorage substo = new SubStorage(prefix, this.asto);
+        substo.save(new Key.From("package"), new Content.From(data)).join();
+        MatcherAssert.assertThat(
+            "Returns storage item with prefix",
+            new BlockingStorage(substo).value(new Key.From("package")),
+            new IsEqual<>(data)
+        );
+        MatcherAssert.assertThat(
+            "Returns storage item with ROOT prefix",
+            new BlockingStorage(new SubStorage(Key.ROOT, this.asto))
+                .value(new Key.From(prefix, "package")),
+            new IsEqual<>(data)
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"my-project", "com/example"})
+    void movesValue(final String pref) {
+        final Key prefix = new Key.From(pref);
+        final byte[] data = "source".getBytes(StandardCharsets.UTF_8);
+        final Key source = new Key.From("src");
+        final Key destination = new Key.From("dest");
+        this.asto.save(new Key.From(prefix, source), new Content.From(data)).join();
+        this.asto.save(
+            new Key.From(prefix, destination),
+            new Content.From("destination".getBytes(StandardCharsets.UTF_8))
+        ).join();
+        final SubStorage substo = new SubStorage(prefix, this.asto);
+        substo.move(source, destination).join();
+        MatcherAssert.assertThat(
+            "Moves key value with prefix",
+            new BlockingStorage(substo).value(destination),
+            new IsEqual<>(data)
+        );
```

Comment:
About the same idea here: use original storage in the assertion, `SubStorage` is being tested here, we do not know if any method works correctly